### ISSUE TITLE
'Upload Error'

### ISSUE THREAD

**Issue by Velheka** [rejetto/hfs#117]
**Describe the bug**
When attempting to upload files to a virtual files system, I get the error 'Upload: Error'
Another potentially related issue is that it also reports that the file is 'too large' for anything larger than a dozen or so KB. This happens for all drives virtual or otherwise despite there being plenty of space on all of them.

**To Reproduce**
Attempt upload to virtualfile system. 'Too large' error happens on both virtual file system and normal disks for files larger than about 50KB

**Environment (please complete the following information):**
 Debian Bullseye

**Comment by rejetto** [comment]
I just made a test uploading a 175MB file, without any problem.
So, there must be something different on your side. You should provide more information.
Are you using this option perhaps?
<img width="254" alt="image" src="https://user-images.githubusercontent.com/1367199/221053515-069efa60-008b-4e3f-a3f5-0086b1dafe0b.png">

Consider providing screenshots, but a video would be better.
Also, having a look at your config file may really help.

**Comment by Velheka** [comment]
Hi Rejetto, thanks for replying.

I did play around with that option to see if it helped. I tried it at 100MB, 0MB, and 1000 but it didn't seem to make a difference. There is plenty more space than 1000MB on the drives.

[Here's a video of me attempting to upload two files to the virtual file system](https://youtu.be/KioA_hlnOs4). The smaller .txt file gets the 'upload error' message, while the larger one fails because it's too large. I'm assuming virtual files are hosted in the home directory of the program running hfs, which has plenty of space.

[Here's another video of me attempting to upload the two files to a disk on the drive](https://youtu.be/o0eP1sTIN0Y). For this one the smaller file uploads just fine but the larger one again fails due to being too large. Again that disk has plenty of space.

**Comment by rejetto** [comment]
thanks for the (useful) videos.

first video: the bug is that you get the upload button, while you should not! (fixed in next release)
In facts, if you read what's written behind the upload field, you'll see "works only on folders with source", and yours doesn't have a source.
<img width="151" alt="image" src="https://user-images.githubusercontent.com/1367199/221180365-f47844c6-6d8f-4f08-ae0f-69c2bd5491d5.png">

second video:  i cannot reproduce the problem, i get no error unless i set a "min. disk" to a very high value, where the upload is rightly refused.
1. Are you totally sure that this video was made with a proper value of "min. disk" ? In your video you don't show the value. Could you double check?
2. Is that Linux? What distribution/version? I just tried with Ubuntu 22 and got on problems, both with min and without.
3. can you make an upload test on another folder and/or another machine ?

**Comment by Velheka** [comment]
Well it's good to rule one of them out! Thankfully I've got no need to upload into virtual folders anyway so it's no big deal for me. For the second video:

1. [For all of those tests the min. disk was set to 100mb](https://i.imgur.com/cGLDAIN.png). I did try with it set to 0 just to see if it made any difference but got the same results. 2
2. Yes, [Debian 11 Bullseye](https://i.imgur.com/tthOVip.png)
3. Another machine would be tricky, but I could probably set one up if we can't work anything else out! I've just tried another upload attempt with a fresh folder in root with chmod 777 permissions, just to rule out if it's actually a folder permissions issue being reported as a size error, but got the same results: small file works, large file gets 'Upload error: file too large'.

**Comment by rejetto** [comment]
the upload to virtual folder is just not possible because i need to know where to store the uploaded files on disk. 😄

about second video, is it possible that you have some sort of proxy in front of you (nginx)? they often need to be accept larger uploads.

If you don't know, I can think of 2 ways to detect it:

1) easier but not very reliable: look into network requests of your browser and see if headers match mine
<img width="400" alt="image" src="https://user-images.githubusercontent.com/1367199/221241628-fc129cc0-2fc9-4134-b309-fecac65b3beb.png">

be sure to find the POST request with the "notificationChannel", then post the screenshot here so I can look at it.

2) better method, i guess, use curl with a "BIG_FILE" like this:
`curl -F upload=@BIG_FILE http://your-domain.com/wrong_folder/`

since wrong_folder doesn't exist, HFS should reply with 404, but if there is a proxy in front it will reply with 413 (too large) because it doesn't know the folder doesn't exist.

**Comment by Velheka** [comment]
I am using Nginx! I couldn't get much results with your first method, and weirdly running a curl upload `curl -F upload=@C:\Users\Me\Downloads\FileZilla_3.62.2_win64-setup.exe http://stuxserver.com:4200/wrongfolder` correctly notices that the location doesn't exist, and pointing it to the open folder and uploading actually works! Still, the Nginx idea sounded too good to ignore so I looked it up, and lo and behold the maximum default filesize for nginx is 1MB, enough for the text file, smaller than the larger file. After adding `client_max_body_size 50G` to the nginx config for that address the large file uploads! So it was an Nginx problem after all.

Thanks a lot for the help Rejetto - I've been the windows version of HFS since 2017  and I can't tell you how relieved I was to stumble into a version of it that I could run on a CLI Linux machine after I finally changed OS. Keep up the good work!
